app/assets/javascripts/components/reviews/fullreview.js.jsx

```diff
@@ -0,0 +1,59 @@
+var FullReview = React.createClass({
+  upVote: function(e){
+    e.preventDefault();
+    this.updateVotes({voteValue: 1, reviewId: this.props.data.id})
+  },
+  downVote: function(e){
+    e.preventDefault();
+    this.updateVotes({voteValue: -1, reviewId: this.props.data.id})
+  },
+  updateVotes: function(voteInfo){
+    $.ajax({
+      url: '/votes',
+      dataType: 'json',
+      method: 'POST',
+      data: voteInfo,
+      success: function(reviews){
+        //remove user's ability to add additional votes to this review
+      }.bind(this),
+      error: function(xhr,status,err){
+        // tell user that vote could not be saved for some reason
+      }.bind(this)
+    })
+  },
+  updateComments: function(){
+    debugger;
+    this.loadCommentsFromServer()
+  },
+  loadCommentsFromServer: function(){
+    // ajax call to grab new comment and append it to state then refresh page? this.props.comments has full list of comments. add new comment to this list.
+  },
+  render: function(){
+    var reviewContent = this.props.data.content
```

Comment:
Is this a component or data? How it is passed in under `this.props.data` is confusing me.